addons/rating/models/rating.py

```diff
@@ -0,0 +1,146 @@
+            record.res_name = name and name[0][1] or ('%s/%s') % (record.res_model, record.res_id)
+
+    @api.model
+    def new_access_token(self):
+        return uuid.uuid4().hex
+
+    res_name = fields.Char(compute='_compute_res_name', string='Resource Name', store=True, help="The name of the rated ressource.")
+    res_model = fields.Char(string='Resource Model', required=True, help="Model name of the rated object", index=True)
+    res_id = fields.Integer(string='Resource ID', required=True, help="Identifier of the rated object", index=True)
+    rated_partner_id = fields.Many2one('res.partner', string="Rated Partner", help="Owner of the rated ressource")
+    partner_id = fields.Many2one('res.partner', string='Customer', help="Author of the rating")
+    rating = fields.Float(string="Rating", group_operator="avg", default=-1, help="Rating value")
+    feedback = fields.Text('Feedback reason', help="Reason of the rating")
+    access_token = fields.Char(string='Security Token', default=new_access_token, help="Access token to set the rating of the value")
+
+    @api.model
+    def apply_rating(self, rate, res_model=None, res_id=None, token=None):
+        """ apply a rating for given res_model/res_id or token. If the res_model is a mail.thread
+            object, a message will be post in the chatter.
+            :param float rating : the rating value to apply
+            :param list res_id : id of the rated object.
+            :param string res_model : name of model.
+            :param string token : access token
+            :return the rating record
+        """
+        domain = [('access_token', '=', token)] if token else [('res_model', '=', res_model), ('res_id', '=', res_id)]
+        rating = self.search(domain, limit=1)
+        if rating:
+            rating.write({'rating' : rate})
+            if hasattr(self.env[rating.res_model], 'message_post'):
+                record = self.env[rating.res_model].sudo().browse(rating.res_id)
+                record.message_post(
+                    body="%s %s <br/><img src='rating/static/src/img/rating_%s.png' style='width:20px;height:20px'/>"
+                    % (rating.sudo().partner_id.name, _('rated it'), rate))
+        return rating
+
+    @api.multi
+    def reset(self):
+        for record in self:
+            record.write({
+                'rating': -1,
+                'access_token': record.new_access_token(),
+                'feedback' : False
+            })
+
+
+
+class RatingMixin(models.AbstractModel):
+
+    _name = 'rating.mixin'
+    _description = "Rating Mixin"
+
+    rating_ids = fields.One2many('rating.rating', 'res_id', string='Rating', domain=lambda self: [('res_model', '=', self._name)])
+
+    @api.multi
+    def rating_send_request(self, template, partner_id, rated_partner_id, reuse_rating=True):
+        """ This method create (empty) rating objects for the current recordsets and send this request by mail (given email
```

Comment:
While linting, you could format the docstring : try to limit comments to 80 characters, :param ... : can be unindented (aligned with the beginning of the string mark). If you want to be complete, I think the correct way to present parameters is
 - :param template: explanation
 - :type template: EmailTemplate record

And: :postal_horn:  !